projects/fundamental-ngx/src/lib/calendar/calendar.component.ts

```diff
@@ -0,0 +1,485 @@
+        if (this.showCalendarMonths) {
+            this.showCalendarMonths = false;
+            this.showCalendarYears = true;
+            this.showCalendarDates = false;
+        } else {
+            this.showCalendarYears = !this.showCalendarYears;
+            this.showCalendarMonths = false;
+            this.showCalendarDates = !this.showCalendarDates;
+        }
+    }
+
+    @HostListener('document:keydown.escape', [])
+    onEscapeKeydownHandler() {
+        this.showCalendarDates = true;
+        this.showCalendarMonths = false;
+        this.showCalendarYears = false;
+    }
+
+    @HostListener('document:click', ['$event'])
+    onClickHandler(e: MouseEvent) {
+        const target = e.target;
+        if (!this.eRef.nativeElement.contains(target)) {
+            this.showCalendarDates = true;
+            this.showCalendarMonths = false;
+            this.showCalendarYears = false;
+        }
+    }
+
+    validateDateFromDatePicker(date: Array<number>): boolean {
+        let isInvalid: boolean = false;
+        let month = date[0];
+        let day = date[1];
+        let year = date[2];
+        let numOfDaysInMonth = 0;
+
+        if (isNaN(month) || isNaN(day) || isNaN(year)) {
+            isInvalid = true;
+            return isInvalid;
+        }
+
+        if (year < 1000 || year > 3000 || month < 1 || month > 12) {
+            isInvalid = true;
+            return isInvalid;
+        } else {
+            numOfDaysInMonth = this.daysPerMonth[month - 1];
+            if (day < 1 || day > numOfDaysInMonth) {
+                isInvalid = true;
+                return isInvalid;
+            }
+        }
+        return isInvalid;
+    }
+
+    ngOnChanges(changes: SimpleChanges) {
+        if(changes.dateFromDatePicker) {
+            console.log(changes.dateFromDatePicker);
+            if (changes.dateFromDatePicker.currentValue.length > 0) {
+                let dateFromDatePickerInput = changes.dateFromDatePicker.currentValue;
+    
+                if (this.datePickerType === 'single') {
+                    let singleDate = dateFromDatePickerInput.replace(/\s/g, '');
+                    singleDate = singleDate.split(/[/]+/);
+    
+                    this.invalidDate = this.validateDateFromDatePicker(singleDate);
+    
+                    if (!this.invalidDate) {
+                        this.selectedDay.date = new Date(singleDate[2], singleDate[0] - 1, singleDate[1]);
+                        this.date = new Date(singleDate[2], singleDate[0] - 1, singleDate[1]);
+                        this.year = this.date.getFullYear();
+                        this.month = this.date.getMonth();
+                        this.monthName = this.monthsFullName[this.date.getMonth()];
+                        this.isInvalidDateInput.emit(this.invalidDate);
+                        this.constructCalendar();
+                    } else {
+                        this.isInvalidDateInput.emit(this.invalidDate);
```

Comment:
I think if an invalid date is typed in to the input, the input should clear and display the proper format as the placeholder like with the time picker

Reply:
Agreed for clearing the input.

Reply:
if an invalid value is typed in the input, on blur event or close for calendar, will clear the input field, and also the selection in the calendar